src/Angor/Avalonia/Angor.Contexts.Funding/Projects/Infrastructure/Impl/DocumentProjectRepository.cs

```diff
@@ -0,0 +1,177 @@
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using Angor.Contexts.Funding.Projects.Domain;
+using Angor.Contexts.Funding.Shared;
+using Angor.Data.Documents.Interfaces;
+using Angor.Shared.Models;
+using Angor.Shared.Services;
+using CSharpFunctionalExtensions;
+using Stage = Angor.Contexts.Funding.Projects.Domain.Stage;
+
+namespace Angor.Contexts.Funding.Projects.Infrastructure.Impl;
+
+public class DocumentProjectRepository(IGenericDocumentCollection<Project> collection, IRelayService relayService,
+    IIndexerService indexerService) : IProjectRepository
+{
+
+    public Task<Result<Project>> GetAsync(ProjectId id)
+    {
+        return GetAllAsync( id )
+            .Map(x =>x.First());
+    }
+
+    public async Task<Result<IEnumerable<Project>>> GetAllAsync(params ProjectId[] ids)
+    {
+        if (ids == null || ids.Length == 0)
+            return Result.Failure<IEnumerable<Project>>("ProjectId cannot be null");
+
+        try
+        {
+            var stringIds = ids.Select(id => id.Value).ToArray();
+
+            var projectResult = await collection.FindAsync(p => stringIds.Any(id => id == p.Id.Value));
+            
+            var localLookup = projectResult.IsSuccess && projectResult.Value.Any() //check the results from the local database
+                ? projectResult.Value.Select(item => item).ToList() : [];
+
+            if (ids.Length == localLookup.Count)
+                return Result.Success(localLookup.AsEnumerable()); //all ids are in the local database, return them
+            
+            var tasks = stringIds
+                .Except(localLookup.Select(p => p.Id.Value)) //ids that are not in the local database
+                .Select(id => Result.Try(() => indexerService.GetProjectByIdAsync(id)));
```

Comment:
we have to try to get rid of the dependency on the custom endpoints, I guess this can be hidden later inside `GetProjectByIdAsync`?